src/components/slider/slider.tsx

```diff
@@ -0,0 +1,433 @@
+import { useTheme } from 'native-base';
+import React, { useState, useRef, useEffect, useMemo } from 'react';
+import { View, Text, PanResponder, Animated, LayoutChangeEvent } from 'react-native';
+
+import { SliderProps } from '../../types/slider';
+import { Input } from '../inputs';
+
+import { useSliderStyles } from './slider.styles';
+
+enum BubbleHandle {
+  SINGLE = 'single',
+  LEFT = 'left',
+  RIGHT = 'right',
+}
+
+enum SubmitType {
+  MIN = 'min',
+  MAX = 'max',
+}
+
+const Slider = ({
+  value = 0,
+  step = 1,
+  lowerLimit = 0,
+  upperLimit = 100,
+  onValueChange,
+  isRange = false,
+  showEndMarkers = false,
+  internalMarkerStep,
+  showValueInput = false,
+  disabled = false,
+}: SliderProps) => {
+  const { sizes } = useTheme();
+  const handleSize = Number(sizes['5']);
+  const [trackLength, setTrackLength] = useState(0);
+
+  const styles = useSliderStyles();
+
+  const calculatePositionFromValue = (val: number) =>
```

Comment:
1. Consider using `useCallback` to cache function definition here and other places.
2. Define return type here, and other places